packages/e2e/axe-selenium/src/main/java/org/talend/axeselenium/axe/Axe.java

```diff
@@ -0,0 +1,146 @@
+        runAxeScript();
+        return run(element, options);
+    }
+
+    /**
+     * Run text-alternative category scan.
+     * @return the scan result JSON object.
+     */
+    public JSONObject runNonTextContent() {
+        return runNonTextContent(null);
+    }
+
+    /**
+     * Run text-alternative category scan on a specific element.
+     * @return the scan result JSON object.
+     */
+    public JSONObject runNonTextContent(final WebElement element) {
+        runAxeScript();
+
+        final Set<String> rules = new HashSet<>();
+        rules.add("label");
+        rules.add("button-name");
+        rules.add("link-name");
+        rules.add("hidden-content");
+        rules.addAll(getRules("cat.name-role-value"));
+        rules.addAll(getRules("cat.text-alternatives"));
+
+        final Options options = Options.builder()
+                .runOnly("rule", rules.toArray(new String[rules.size()]))
+                .build();
+        return run(element, options);
+    }
+
+    /**
+     * Run a scan on the element (or document if not provided)
+     * @param element the element to scan. "document" is scanned if element is null.
+     * @param options axe-core options.
+     * @return the scan result JSON object.
+     */
+    public JSONObject run(final WebElement element, final Options options) {
+        final String axeOptions = options == null ? "{}" : options.toString();
+        final String command = String.format(
+                "const done = arguments[arguments.length - 1];\n" +
+                "function handleResult(error, report) { if (error) { throw error; } return done(report); }\n" +
+                "const targetElement = arguments[0] || document;\n" +
+                "axe.run(targetElement, %s, handleResult);",
+                axeOptions
+        );
+        final Object response = ((JavascriptExecutor) this.driver).executeAsyncScript(command, element);
+
+        return new JSONObject((Map) response);
+    }
+
+    /**
+     * Save scan result in ${reportFolder}/${name}.json.
+     * Build a junit friendly violation report.
+     * @param reportFolder The folder where to store the scan result.
+     * @param name The result file name.
+     * @param result The scan result.
+     * @return the formatted violation report string, null if no violation.
+     */
+    public String reportViolations(final String reportFolder, final String name, final JSONObject result) {
+        new File(reportFolder).mkdirs();
+        final JSONArray violations = result.getJSONArray("violations");
+        if (violations.length() > 0) {
+            final String formattedViolations = reporter.report(violations);
+            reporter.writeResults(reportFolder + '/' + name, result);
+            reporter.writeResults(reportFolder + '/' + name, formattedViolations);
+            return formattedViolations;
+        }
+        return null;
```

Comment:
I'd rather see `""` being returned to keep the returned type consistent.

Reply:
It's an indicator that there are no violation. `violations == null` vs `violations.equals("")`.
I let the method return null.